framework/src/org/checkerframework/common/wholeprograminference/WholeProgramInferenceScenesHelper.java

```diff
@@ -248,13 +249,13 @@ private void removeIgnoredAnnosFromScene(AScene scene) {
      */
     private void removeIgnoredAnnosFromATypeElement(ATypeElement typeEl, TypeUseLocation loc) {
         Set<Annotation> annosToRemove = new HashSet<>();
-        Set<String> annosToIgnoreForLocation = annosToIgnore.get(loc);
+        Set<String> annosToIgnoreForLocation = annosToIgnore.get(Pair.of(typeEl.toString(), loc));
         if (annosToIgnoreForLocation == null) {
             // No annotations to ignore for that position.
             return;
         }
         for (Annotation anno : typeEl.tlAnnotationsHere) {
-            if (annosToIgnoreForLocation.contains(anno.def().name)) {
+            if (annosToIgnoreForLocation.contains(anno.def().toString())) {
```

Comment:
This change seems independent of the others in the pull request.  I'm not saying it's wrong, but I'm wondering whether it is an independent bug fix or is something that is related to the rest of the pull request.